conduit/pipeline.go

```diff
@@ -0,0 +1,242 @@
+package conduit
+
+import (
+	"context"
+	"fmt"
+
+	log "github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v3"
+	"io/ioutil"
+	"path/filepath"
+
+	"github.com/algorand/go-algorand/util"
+	"github.com/algorand/indexer/data"
+	"github.com/algorand/indexer/exporters"
+	"github.com/algorand/indexer/importers"
+	"github.com/algorand/indexer/processors"
+)
+
+const autoLoadParameterConfigName = "conduit.yml"
+
+// PipelineConfig stores configuration specific to the conduit pipeline
+type PipelineConfig struct {
+	ConduitConfig *Config
+	// Store a local copy to access parent variables
+	Importer   map[string]interface{}   `yaml:"Importer"`
+	Processors []map[string]interface{} `yaml:"Processors"`
+	Exporter   map[string]interface{}   `yaml:"Exporter"`
+}
+
+// Valid validates pipeline config
+func (cfg *PipelineConfig) Valid() error {
+	if cfg.ConduitConfig == nil {
+		return fmt.Errorf("conduit configuration was nil")
+	}
+
+	if len(cfg.Importer) == 0 {
```

Comment:
```suggestion
	if len(cfg.Importer) != 1 {
```

Reply:
Isn't this the opposite logic though?  We are trying to see if there are no importers and then throw an error.

Reply:
It's just a matter of what we want to consider a valid pipeline. I think requiring a single importer and Exporter makes sense, and having 0 or more Processors is also valid. So we want to check that they have exactly 1 of Importer/Exporter, and we don't need to validate length of processors because any number is valid.

Reply:
Discussed out of band--the existing logic is what we want.